caikit/runtime/service_factory.py

```diff
@@ -130,9 +136,27 @@ def get_service_package(cls, service_type: ServiceType) -> ServicePackage:
         # Then do API introspection to come up with all the API definitions to support
         caikit_config = get_config()
         clean_modules = ServicePackageFactory._get_and_filter_modules(
-            caikit_config, caikit_config.runtime.library
+            caikit_config, caikit_config.runtime.library, write_modules_file
         )
 
+        # Assert for backwards compatibility, if enabled
+        backwards_compat_conf = (
+            caikit_config.runtime.service_generation.backwards_compatibility
+        )
+        if backwards_compat_conf and backwards_compat_conf.enabled:
```

Comment:
Can we stick this backwards compatibility checking logic into a private method to keep this public one as concise as possible?